docs(quick-start): add Run a Fiber Node overview (#73)
nervosnetwork/fiber-docs: content/docs/quick-start/run-a-node/index.mdx

## Node Forms

Fiber ships in two forms. The right one depends on where you want to run it and who controls the runtime.

### Native Node (`fnn`)

The native Rust binary is the full production node. It runs as a standalone process on a server or local machine, stores channel state locally (RocksDB or SQLite), and exposes JSON-RPC on `127.0.0.1:8227` by default.

| Aspect | Details |
|--------|---------|
| **Runtime** | Native binary (Linux, macOS, Windows) |
| **Best for** | Routing nodes, production services, server deployments |
| **Storage** | Local RocksDB/SQLite database |
| **Network exposure** | Public IP recommended for routing; private nodes can connect through public relay nodes |
| **Interaction** | `fnn-cli` or raw JSON-RPC |

Use the native node when you need full control, persistent storage, and the ability to accept inbound channel requests from the public internet.

→ [Run a Native Node](/docs/quick-start/run-a-node/rust)

> ### WASM Node (`fiber-wasm` / `fiber-js`)

Reviewer: ```suggestion
### WASM Node (`fiber-js`)
```

> `fiber-wasm` compiles the same node logic to WebAssembly so it can run inside a browser or a `Node.js` process. It stores data in IndexedDB (via a WASM worker) and is designed for client-side applications that do not want to operate a backend server.

Reviewer: ```suggestion
`fiber-js` compiles the same node logic to WebAssembly so it can run inside a browser or a `Node.js` process. It stores data in IndexedDB (via a WASM worker) and is designed for client-side applications that do not want to operate a backend server.
```

| Aspect | Details |
|--------|---------|
| **Runtime** | Browser or `Node.js` |
| **Best for** | Browser wallets, web games, client-side dApps |
| **Storage** | IndexedDB |
| **Network exposure** | No public IP required; can connect through relay nodes |

> | **Interaction** | JavaScript/TypeScript API (`fiber-wasm` package) |

Reviewer: ```suggestion
| **Interaction** | JavaScript/TypeScript API (`fiber-js` package) |
```

- Accept inbound channels from users.
- Route multi-hop payments and earn fees.
- Provide liquidity as a service.

You will need:

- A stable server with a public IP.
- A trusted CKB RPC endpoint (especially important on Mainnet).
- Sufficient CKB/UDT liquidity to fund and maintain channels.
- (Optional) A TLS proxy if you also want browser/WASM clients to reach you over WSS.

→ See [WSS Configuration](https://github.com/nervosnetwork/fiber/blob/v0.9.0-rc3/docs/fiber-node-wss.md) for browser-facing deployments.

### Private Node Behind Relay

Not every node has to be public. A private node can open outbound channels to public relay nodes and then send or receive payments through those relays. This is the easiest way to participate without exposing a public IP or running server infrastructure.

This pattern is covered in the [Public Nodes User Manual](https://github.com/nervosnetwork/fiber/blob/v0.9.0-rc3/docs/public-nodes.md) and is the default mental model for the WASM node.

### Browser-Embedded Node

> With `fiber-wasm`, a Fiber node can live inside a web page. Application code creates a `Fiber` instance and starts it with `fiber.start(config, fiberKeyPair, ckbSecretKey, ...)`; the node can then connect to Testnet bootnodes over WebSocket. This is ideal for wallets, games, and demos where installation friction must be minimal.

Reviewer: ```suggestion
With `fiber-js`, a Fiber node can live inside a web page. Application code creates a `Fiber` instance and starts it with `fiber.start(config, fiberKeyPair, ckbSecretKey, ...)`; the node can then connect to Testnet bootnodes over WebSocket. This is ideal for wallets, games, and demos where installation friction must be minimal.
```

### Local / Testnet Node

The fastest way to experiment. You run `fnn` locally against Fiber Testnet, fund the address from a testnet faucet, and open channels with the public Testnet relay nodes. No public IP is required because you initiate outbound connections to the public nodes.

Typical flow:

1. Download or build the `fnn` binary.
2. Generate or export a CKB private key.
3. Copy `config/testnet/config.yml` and set `FIBER_SECRET_KEY_PASSWORD`.
4. Start the node and connect to a public Testnet node.
5. Open a channel and try [Basic Transfer](/docs/quick-start/basic-transfer).

### Production Public Node

A public node has a reachable P2P address (default port `8228`) and is advertised in the network graph. Other nodes can connect to it and open channels. This is the form you want if you intend to:

- Accept inbound channels from users.
- Route multi-hop payments and earn fees.
- Provide liquidity as a service.

You will need:

- A stable server with a public IP.
- A trusted CKB RPC endpoint (especially important on Mainnet).
- Sufficient CKB/UDT liquidity to fund and maintain channels.
- (Optional) A TLS proxy if you also want browser/WASM clients to reach you over WSS.

> → See [WSS Configuration](https://github.com/nervosnetwork/fiber/blob/main/docs/fiber-node-wss.md) for browser-facing deployments.

Reviewer 2: This link returns 404

### Private Node Behind Relay

Not every node has to be public. A private node can open outbound channels to public relay nodes and then send or receive payments through those relays. This is the easiest way to participate without exposing a public IP or running server infrastructure.

This pattern is covered in the [Public Nodes User Manual](https://github.com/nervosnetwork/fiber/blob/main/docs/public-nodes.md) and is the default mental model for the WASM node.

### Browser-Embedded Node

> With `fiber-wasm`, a Fiber node can live inside a web page. The user starts the node by calling `fiber(config)` in JavaScript, and the node connects to Testnet bootnodes over WebSocket. This is ideal for wallets, games, and demos where installation friction must be minimal.

Reviewer 2: ```suggestion
With `fiber-wasm`, a Fiber node can live inside a web page. Application code creates a `Fiber` instance and starts it with `fiber.start(config, fiberKeyPair, ckbSecretKey, ...)`; the node can then connect to Testnet bootnodes over WebSocket. This is ideal for wallets, games, and demos where installation friction must be minimal.
```

→ [Run a WASM Node](/docs/quick-start/run-a-node/fiberjs)

## Mainnet vs. Testnet

Fiber supports both Mainnet and Testnet deployments. The network is selected in `config.yml` via the `fiber.chain` field.

| Network | Purpose | CKB RPC Default | Notes |
|---------|---------|-----------------|-------|
| **Testnet** | Development and experimentation | `https://testnet.ckbapp.dev/` | Free testnet CKB and RUSD available from faucets |
| **Mainnet** | Production value transfer | Your own trusted CKB node or a trusted RPC provider | Public nodes exist, but always verify liquidity and policies |

Testnet is the recommended place to start. Once you are comfortable with channel management, payment flows, and backup procedures, move the same setup to Mainnet.

## What You Need Before Starting

Regardless of the form you choose, running a node requires:

> 1. **A CKB private key** — used to sign funding and settlement transactions. FNN stores it encrypted under `FIBER_SECRET_KEY_PASSWORD`.

Reviewer 2: ```suggestion
1. **A CKB private key** — Native `fnn` uses a CKB private key file protected by `FIBER_SECRET_KEY_PASSWORD`; WASM/fiber-js receives key material through the JS API.
```

2. **A CKB RPC endpoint** — the node reads chain state and broadcasts transactions through it.
3. **A `config.yml`** — selects the network, bootnodes, listening addresses, RPC port, and whitelisted UDTs.

> 4. **Enough CKB for on-chain fees and channel reserve** — opening a channel requires funding CKB on-chain, and each side reserves 99 CKB for the commitment lock and shutdown fee.

Reviewer 2: Could we add one sentence per prereq explaining where to obtain it? For example, CKB private key via `ckb-cli`, CKB RPC from the bundled Testnet default or a trusted Mainnet provider, `config.yml` from config/testnet or config/mainnet, and CKB from a Testnet faucet or Mainnet wallet.

---

> title: "Run a Fiber Node: Overview"

Reviewer 3: 直接叫 Overview 怎么样

date: 2026-06-13
---

import { Callout } from 'fumadocs-ui/components/callout';

A **Fiber node** is a running instance of the [Fiber Network Node (FNN)](https://github.com/nervosnetwork/fiber) — the reference implementation of the Fiber Network Protocol (FNP). In practical terms, it is the software that connects you to the Fiber peer-to-peer payment network, manages payment channels, routes multi-hop payments, and signs the on-chain transactions that anchor channel state to CKB.

If you want to **send or receive** payments over Fiber, build a wallet that speaks the protocol, or **earn routing fees** by forwarding payments for others, you need a node.

<Callout type="info">
FNN is still under active development. Before upgrading between versions, close your channels or migrate storage carefully — protocol and storage formats can change between releases.
</Callout>

## What a Fiber Node Does

Running a node gives you a local view of the Fiber network and the ability to:

- **Maintain payment channels** — open, update, and close channels with other nodes.
- **Send and receive payments** — both direct peer payments and multi-hop routed payments.

> - **Swap assets** — move value across CKB-native assets (CKB, UDTs such as USDI/RUSD), and across networks (e.g., between Lightning and Fiber via cross-chain constructs).

Reviewer 3: 目前只有 USDI 了

### Local / Testnet Node

The fastest way to experiment. You run `fnn` locally against Fiber Testnet, fund the address from a testnet faucet, and open channels with the public Testnet relay nodes. No public IP is required because you initiate outbound connections to the public nodes.

Typical flow:

1. Download or build the `fnn` binary.
2. Generate or export a CKB private key.
3. Copy `config/testnet/config.yml` and set `FIBER_SECRET_KEY_PASSWORD`.
4. Start the node and connect to a public Testnet node.
5. Open a channel and try [Basic Transfer](/docs/quick-start/basic-transfer).

### Production Public Node

A public node has a reachable P2P address (default port `8228`) and is advertised in the network graph. Other nodes can connect to it and open channels. This is the form you want if you intend to:

- Accept inbound channels from users.
- Route multi-hop payments and earn fees.
- Provide liquidity as a service.

You will need:

- A stable server with a public IP.
- A trusted CKB RPC endpoint (especially important on Mainnet).
- Sufficient CKB/UDT liquidity to fund and maintain channels.
- (Optional) A TLS proxy if you also want browser/WASM clients to reach you over WSS.

> → See [WSS Configuration](https://github.com/nervosnetwork/fiber/blob/develop/docs/fiber-node-wss.md) for browser-facing deployments.

Reviewer 3: 404

### Browser-Embedded Node

With `fiber-wasm`, a Fiber node can live inside a web page. Application code creates a `Fiber` instance and starts it with `fiber.start(config, fiberKeyPair, ckbSecretKey, ...)`; the node can then connect to Testnet bootnodes over WebSocket. This is ideal for wallets, games, and demos where installation friction must be minimal.

→ [Run a WASM Node](/docs/quick-start/run-a-node/fiberjs)

## Mainnet vs. Testnet

Fiber supports both Mainnet and Testnet deployments. The network is selected in `config.yml` via the `fiber.chain` field.

| Network | Purpose | CKB RPC Default | Notes |
|---------|---------|-----------------|-------|
| **Testnet** | Development and experimentation | `https://testnet.ckbapp.dev/` | Free testnet CKB and RUSD available from faucets |
| **Mainnet** | Production value transfer | Your own trusted CKB node or a trusted RPC provider | Public nodes exist, but always verify liquidity and policies |

Testnet is the recommended place to start. Once you are comfortable with channel management, payment flows, and backup procedures, move the same setup to Mainnet.

> ## What You Need Before Starting

Reviewer 3: 我感觉放到下面的 node /wasm 里怎么样，另外 2、 3 、 4  感觉都不是用户主观上必须的，第一点可以用 ckb-cli

Regardless of the form you choose, running a node requires:

1. **A CKB private key** — Native `fnn` uses a CKB private key file protected by `FIBER_SECRET_KEY_PASSWORD`; WASM/`fiber-js` receives key material through the JS API.
   - Generate one with [`ckb-cli`](https://github.com/nervosnetwork/ckb-cli) or export it from a CKB wallet. Native `fnn` expects a hex key placed at ``<ckb_base_dir>/key`` and will migrate it to an encrypted file on first start.
2. **A CKB RPC endpoint** — the node reads chain state and broadcasts transactions through it.
   - For Testnet, the sample `config/testnet/config.yml` uses `https://testnet.ckbapp.dev/`. For Mainnet, run your own CKB node or use a trusted RPC provider.
3. **A `config.yml`** — selects the network, bootnodes, listening addresses, RPC port, and whitelisted UDTs.
   - Start from [`config/testnet/config.yml`](https://github.com/nervosnetwork/fiber/blob/develop/config/testnet/config.yml) or [`config/mainnet/config.yml`](https://github.com/nervosnetwork/fiber/blob/develop/config/mainnet/config.yml) in the Fiber repo and adjust the values for your deployment.

> 4. **Enough CKB for on-chain fees and channel reserve** — opening a channel requires funding CKB on-chain, and each side reserves 99 CKB for the commitment lock and shutdown fee.

Reviewer 3: 这里如果是单向通道，其实卖家不用存 ckb 也行
